designer/client/src/components/graph/node-modal/aggregate/aggMapLikeParser.tsx

```diff
@@ -1,236 +1,57 @@
+        return this.#parseList(input, ast);
+    };
+
+    parseObject = (input: string) => {
+        const ast = this.#parse(input);
+        return this.#parseObject(input, ast);
+    };
+
+    #parse(input: string) {
+        try {
+            return parse(input, true);
+        } catch (e) {
             return null;
         }
-        obj[lit] = value;
-        return obj;
-    });
-    private mapItems = this.RULE("mapItems", () => {
-        const obj = {};
-        this.MANY_SEP({
-            SEP: Comma,
-            DEF: () => {
-                Object.assign(obj, this.SUBRULE(this.objectItem));
-            },
-        });
-        return obj;
-    });
-    private aggMap = this.RULE("aggMap", () => {
-        const opening = this.CONSUME(MapOpen);
-        const obj = this.SUBRULE(this.mapItems);
-        this.CONSUME(MapClose);
-
-        if (!opening.image || opening.isInsertedInRecovery) {
-            return null;
-        }
-
-        return obj;
-    });
-    private plainMap = this.RULE("plainMap", () => {
-        const opening = this.CONSUME(LCurly);
-        const obj = this.SUBRULE(this.mapItems);
-        this.CONSUME(RCurly);
-
-        if (!opening.image || opening.isInsertedInRecovery) {
-            return null;
-        }
-
-        return obj;
-    });
-    private object = this.RULE("object", () => {
-        return this.OR([{ ALT: () => this.SUBRULE(this.aggMap) }, { ALT: () => this.SUBRULE(this.plainMap) }]);
-    });
-    private collection = this.RULE("collection", () => {
-        const arr = [];
-        this.CONSUME(CollectionOpen);
-        this.MANY_SEP({
-            SEP: Comma,
-            DEF: () => {
-                const item = this.SUBRULE(this.collectionItem);
-                if (!item) return;
-                arr.push(item);
-            },
-        });
-        this.CONSUME(CollectionClose);
-        return arr;
-    });
-    private list = this.RULE("list", () => {
-        const arr = [];
-        this.CONSUME(LCurly);
-        this.MANY_SEP({
-            SEP: Comma,
-            DEF: () => {
-                const item = this.SUBRULE(this.collectionItem);
-                if (!item) return;
-                arr.push(item);
-            },
-        });
-        this.CONSUME(ListClose);
-        return arr;
-    });
-    private groupBy = this.RULE("groupBy", () => {
-        return this.OR([{ ALT: () => this.SUBRULE(this.collection) }, { ALT: () => this.SUBRULE(this.list) }]);
-    });
-    private fullText = "";
-
-    constructor(tokens = aggMapTokens) {
-        super(tokens, { recoveryEnabled: true });
-        this.lexer = new Lexer(tokens);
-        this.parseObject = withLogs(this.parseObject.bind(this));
-        this.parseList = withLogs(this.parseList.bind(this));
-        this.performSelfAnalysis();
     }
 
-    parseList(input: string): Array<string> {
-        this.tokenizeInput(input);
-        return this.groupBy() || null;
+    #parseList(input: string, ast?: Ast) {
```

Comment:
The same as above - I'd call it parseListToStringExpression or parseAggregateKey